### PR TITLE
Fix check_args macro

### PR DESCRIPTION
**Pull request by andreasnoack**

### CODE REVIEW

**Comment by ararslan** [comment]
The 0.6 Linux failure is because Rmath is being built from source, whereas it's coming from Homebrew on OS X. Looks like maybe BinDeps is the culprit? If it's Rmath, we'll have to bump the version of StatsFuns, since that's how we're calling into Rmath here currently...

**Comment by yuyichao** [comment]
```diff
diff --git a/src/utils.jl b/src/utils.jl
index 58c9cb1..08058df 100644
--- a/src/utils.jl
+++ b/src/utils.jl
@@ -105,9 +105,13 @@ end
 # for checking the input range of quantile functions
 # comparison with NaN is always false, so no explicit check is required
 macro checkquantile(p,ex)
+    p = esc(p)
+    ex = esc(ex)
     :(zero($p) <= $p <= one($p) ? $ex : NaN)
 end
 macro checkinvlogcdf(lp,ex)
+    lp = esc(lp)
+    ex = esc(ex)
     :($lp <= zero($lp) ? $ex : NaN)
 end
```

**Comment by andreasnoack** [comment]
@yuyichao Thanks. I've fixed it.

File: src/utils.jl [hunk 1]
```diff
@@ -105,9 +105,11 @@ end
 # for checking the input range of quantile functions
 # comparison with NaN is always false, so no explicit check is required
 macro checkquantile(p,ex)
+    p = esc(p)
```
Comment (yuyichao) [inline src/utils.jl]: Ex too

**Comment by andreasnoack** [comment]
Done.

File: src/utils.jl [hunk 2]
```diff
@@ -2,9 +2,9 @@
 
 macro check_args(D, cond)
     quote
-        if !($cond)
+        if !($(esc(cond)))
             throw(ArgumentError(string(
-                $(string(D)), ": the condition ", $(string(cond)), " is not satisfied.")))
+                $(string(esc(D))), ": the condition ", $(string(esc(cond))), " is not satisfied.")))
```
Comment (yuyichao) [inline src/utils.jl]: Actually, I don't think you want these two `esc`.

**Comment by yuyichao** [comment]
And just for the reference, the change I was testing locally before I saw this.

```patch
diff --git a/src/utils.jl b/src/utils.jl
index 58c9cb1..08058df 100644
--- a/src/utils.jl
+++ b/src/utils.jl
@@ -2,7 +2,7 @@
 
 macro check_args(D, cond)
     quote
-        if !($cond)
+        if !($(esc(cond)))
             throw(ArgumentError(string(
                 $(string(D)), ": the condition ", $(string(cond)), " is not satisfied.")))
         end
@@ -105,9 +105,13 @@ end
 # for checking the input range of quantile functions
 # comparison with NaN is always false, so no explicit check is required
 macro checkquantile(p,ex)
+    p = esc(p)
+    ex = esc(ex)
     :(zero($p) <= $p <= one($p) ? $ex : NaN)
 end
 macro checkinvlogcdf(lp,ex)
+    lp = esc(lp)
+    ex = esc(ex)
     :($lp <= zero($lp) ? $ex : NaN)
 end
```

**Comment by andreasnoack** [comment]
When https://github.com/JuliaLang/METADATA.jl/pull/7245 is merged, the tests on master can be restarted.

**Pull request merged**